### DISCUSSION TITLE
Mobile file receiver register

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#8316] [Ideas]
I have a humble idea. Currently we only accept dropping to import on desktop, But what if we want to share an image or a website to the AI chatbot plugin on the mobile?

We need to come up with a standard way for plugin to register incoming file handler (import as tiddler, or as external file, or give it to AI).
And a way for mobile APP (maybe Also desktop apps) to send the receipt file to this file handler.

So there can be a modal telling user which way he want the file received to be used.

Just think about this because i want to share what I read with ai and let it decide if it need to record it into my diary.

**Comment by linonetwo** [comment]
Can be done by `$tw.hooks.addHook('th-importing-file'` to trigger a custom UI.

**Comment by Jermolene** [comment]
Hi @linonetwo in MWS I handle imports by POSTing files to `/wiki/foobar/tiddlers`. At the moment, incoming files are just turned into attachments directly, but the intention is to make this the front end for a more elaborate interactive import process.

**Reply by linonetwo** [reply]
When involving AI, not all incoming files are ment to be saved, some are presented to AI for parsing and never saved as tiddler or external files.
